### PR TITLE
add learning rate api of optimizer

### PR DESCRIPTION
**Pull request by phlrain**
add learning rate api of optimizer

image

image

image

### CODE REVIEW

File: python/paddle/fluid/optimizer.py [hunk 1]
```diff
@@ -281,6 +281,38 @@ def _create_global_learning_rate(self):
...
+                with fluid.dygraph.guard():
+                    adam = fluid.optimizer.Adam(0.001)
+                    lr = adam.current_step_lr
+
+        """
+        current_lr = self._global_learning_rate()
+        if current_lr:
+            return self._global_learning_rate().numpy()[0]
+
+        if isinstance(self._learning_rate, (float, int)):
```
Comment (liupluswei) [inline python/paddle/fluid/optimizer.py]: image
从Optimizer初始化代码看的话感觉不需要判断是否为int

File: python/paddle/fluid/optimizer.py [hunk 2]
```diff
@@ -281,6 +281,38 @@ def _create_global_learning_rate(self):
...
+            return self._learning_rate
+        elif isinstance(self._learning_rate, LearningRateDecay):
```
Comment (liupluswei) [inline python/paddle/fluid/optimizer.py]: 单看这块代码，没有else的话会觉得函数存在不返回learning_rate的case
比如self.learning_rate既不是float，也不是LearningRateDecay, 那么函数可能就没返回值

File: python/paddle/fluid/optimizer.py (hunk 1 above)
Comment (phlrain) [reply python/paddle/fluid/optimizer.py]: done

File: python/paddle/fluid/optimizer.py (hunk 2 above)
Comment (phlrain) [reply python/paddle/fluid/optimizer.py]: optimizer 在__init__ 就判定了 learning的类型 为 float 或者 LearningRateDecay 

这块把 elif 直接替换为 else

**Review by liupluswei — DISMISSED** [review]
LGTM

File: python/paddle/fluid/optimizer.py [hunk 3]
```diff
@@ -281,6 +281,38 @@ def _create_global_learning_rate(self):
...
+        Return:
+            learing_rate(float) : The learning rate of the current step.
+
+        Examples:
+            .. code-block:: python
+
+                import paddle.fluid as fluid
+                with fluid.dygraph.guard():
+                    adam = fluid.optimizer.Adam(0.001)
+                    lr = adam.current_step_lr
```
Comment (lanxianghit) [inline python/paddle/fluid/optimizer.py]: 给一下输出结果，另外，因为使用和不使用LearningRateDecay行为不同，最好给出使用和不使用情况下的示例和返回结果

File: python/paddle/fluid/optimizer.py [hunk 4]
```diff
@@ -281,6 +283,39 @@ def _create_global_learning_rate(self):
                 dtype='float32' if self._dtype is None else self._dtype,
                 persistable=True)
 
+    @framework.dygraph_only
```
Comment (lanxianghit) [inline python/paddle/fluid/optimizer.py:288]: dygraph_only的在文档里说明一下吧

File: python/paddle/fluid/optimizer.py (hunk 3 above)
Comment (JepsonWong) [reply python/paddle/fluid/optimizer.py]: done

File: python/paddle/fluid/optimizer.py (hunk 4 above)
Comment (JepsonWong) [reply python/paddle/fluid/optimizer.py:288]: done

File: python/paddle/fluid/optimizer.py [hunk 5]
```diff
@@ -281,6 +283,68 @@ def _create_global_learning_rate(self):
...
+
+        Returns:
+            float: The learning rate of the current step.
+
+        Examples:
+            .. code-block:: python
+
+                import paddle.fluid as fluid
+
+                # example1
```
Comment (lanxianghit) [inline python/paddle/fluid/optimizer.py]: 每个example对应什么情况请做一个简要说明

File: python/paddle/fluid/optimizer.py [hunk 6]
```diff
@@ -281,6 +283,68 @@ def _create_global_learning_rate(self):
                 dtype='float32' if self._dtype is None else self._dtype,
                 persistable=True)
 
+    @framework.dygraph_only
+    def current_step_lr(self):
+        """
+        **Notes**:
```
Comment (lanxianghit) [inline python/paddle/fluid/optimizer.py]: 官网文档现在已经支持Note样式，请使用最新样式，可以参考内部文档：http://wiki.baidu.com/pages/viewpage.action?pageId=902795119

File: python/paddle/fluid/optimizer.py (hunk 5 above)
Comment (JepsonWong) [reply python/paddle/fluid/optimizer.py]: done

File: python/paddle/fluid/optimizer.py (hunk 6 above)
Comment (JepsonWong) [reply python/paddle/fluid/optimizer.py]: done

**Review by lanxianghit — APPROVED** [review]

**Review by saxon-zh — APPROVED** [review]
